tracer/test/Datadog.Trace.Tests/DatabaseMonitoring/DatabaseMonitoringPropagatorTests.cs

```diff
@@ -0,0 +1,54 @@
+// <copyright file="DatabaseMonitoringPropagatorTests.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.DatabaseMonitoring;
+using Datadog.Trace.ExtensionMethods;
+using Xunit;
+
+namespace Datadog.Trace.Tests.DatabaseMonitoring
+{
+    public class DatabaseMonitoringPropagatorTests
+    {
+        [Theory]
+        [InlineData("disabled", SamplingPriority.UserKeep, "")]
+        [InlineData("Service", SamplingPriority.AutoReject, "/*ddps='Test.Service',dddbs='Test.Service-mysql',ddpv='1.0.0',dde='testing'*/")]
+        [InlineData("fUll", SamplingPriority.AutoKeep, "/*ddps='Test.Service',dddbs='Test.Service-mysql',ddpv='1.0.0',dde='testing',traceparent='00-00000000000000006172c1c9a829c71c-05a5f7b5320d6e4d-01'*/")]
+        [InlineData("full", SamplingPriority.UserReject, "/*ddps='Test.Service',dddbs='Test.Service-mysql',ddpv='1.0.0',dde='testing',traceparent='00-00000000000000006172c1c9a829c71c-05a5f7b5320d6e4d-00'*/")]
+        public void ExpectedCommentInjected(string propagationMode, SamplingPriority samplingPriority, string expectedComment)
+        {
+            DbmPropagationLevel dbmPropagationLevel;
+            Enum.TryParse(propagationMode, true, out dbmPropagationLevel);
+
+            Span span = Tracer.Instance.StartSpan("mysql.query", null, null, "Test.Service-mysql", null, 7021887840877922076, 407003698947780173);
```

Comment:
You should probably avoid using `Trace.Instance` since we don't want to use any shared state in unit tests. You could wither create a new `Tracer` instance (you may need to mock some of the dependencies; there are other tests that do this, I can help you find exampled), or you can create the `Span` directly since that's all you need for this test.